### PR TITLE
Migrating test assertions to AssertJ

### PR DESCRIPTION
**Pull request by AnatolyPopov**

### CODE REVIEW

**Review by snuyanzin — DISMISSED** [review]

File: src/integration-test/java/io/aiven/kafka/connect/http/AvroIntegrationTest.java [hunk 1]
```diff
@@ -179,16 +175,12 @@ final void testBasicDelivery() throws ExecutionException, InterruptedException {
...
-        assertIterableEquals(expectedBodies, bodyRecorderHandler.recorderBodies());
 
-        log.info("{} HTTP requests were expected, {} were successfully delivered",
-                expectedBodies.size(),
-                bodyRecorderHandler.recorderBodies().size());
+        await("All expected requests received by HTTP server")
+                .atMost(Duration.ofSeconds(15)).pollInterval(Duration.ofMillis(100))
+                .until(() -> bodyRecorderHandler.recorderBodies().size() >= expectedBodies.size());
+
+        assertThat(bodyRecorderHandler.recorderBodies()).containsExactlyInAnyOrderElementsOf(expectedBodies);
```
Comment (hackaugusto) [inline src/integration-test/java/io/aiven/kafka/connect/http/AvroIntegrationTest.java:183]: question: the previous check seemed more strict, it reads as if the order mattered. Is this change intentional?

File: src/integration-test/java/io/aiven/kafka/connect/http/AvroIntegrationTest.java [hunk 2]
```diff
@@ -179,16 +175,12 @@ final void testBasicDelivery() throws ExecutionException, InterruptedException {
         for (final Future<RecordMetadata> sendFuture : sendFutures) {
             sendFuture.get();
         }
-        TestUtils.waitForCondition(() -> bodyRecorderHandler.recorderBodies().size() >= expectedBodies.size(),
-                15000,
-                "All requests received by HTTP server"
-        );
-        log.info("Recorded request bodies: {}", bodyRecorderHandler.recorderBodies());
```
Comment (hackaugusto) [inline src/integration-test/java/io/aiven/kafka/connect/http/AvroIntegrationTest.java:186]: I suppose this was here for debugging of the test, why remove it? Does the `assertThat` utility provide a better message in case of failure?

File: src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java [hunk 3]
```diff
@@ -175,17 +168,12 @@ void testBasicDelivery() throws ExecutionException, InterruptedException {
...
-        assertIterableEquals(expectedBodies, bodyRecorderHandler.recorderBodies());
+        await("All expected requests received by HTTP server")
+                .atMost(Duration.ofSeconds(15)).pollInterval(Duration.ofMillis(100))
+                .until(() -> bodyRecorderHandler.recorderBodies().size() >= expectedBodies.size());
 
-        log.info("{} HTTP requests were expected, {} were successfully delivered",
-            expectedBodies.size(),
-            bodyRecorderHandler.recorderBodies().size());
+        assertThat(bodyRecorderHandler.recorderBodies())
+                .containsExactlyInAnyOrderElementsOf(expectedBodies);
```
Comment (hackaugusto) [inline src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java:176]: here too the previous test seemed to care about order

File: src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java [hunk 4]
```diff
@@ -175,17 +168,12 @@ void testBasicDelivery() throws ExecutionException, InterruptedException {
             sendFuture.get();
         }
 
-        TestUtils.waitForCondition(
-            () -> bodyRecorderHandler.recorderBodies().size() >= expectedBodies.size(),
-            15000,
-            "All requests received by HTTP server"
-        );
-        log.info("Recorded request bodies: {}", bodyRecorderHandler.recorderBodies());
```
Comment (hackaugusto) [inline src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java:183]: same question about logging the entries for debugging

**Review by hackaugusto — COMMENTED** [review]
there are quite a few changes and I'm not familiar with the API, so I'm not going to approve/request changes.

File: src/integration-test/java/io/aiven/kafka/connect/http/AvroIntegrationTest.java (hunk 1 above)
Comment (AnatolyPopov) [reply src/integration-test/java/io/aiven/kafka/connect/http/AvroIntegrationTest.java:183]: Great catch! You are right! Not sure however whether this is required here. But will fix.

File: src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java (hunk 4 above)
Comment (AnatolyPopov) [reply src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java:183]: The new assertion will show in the fail message what was in this list. Unless the test fails I guess nobody reads the logs, so I don't think it's useful.

File: src/integration-test/java/io/aiven/kafka/connect/http/AvroIntegrationTest.java (hunk 2 above)
Comment (AnatolyPopov) [reply src/integration-test/java/io/aiven/kafka/connect/http/AvroIntegrationTest.java:186]: Yes, AssertJ and it's `assertThat` provide a lot more clear messages. Also in this case I checked how this one is printed and it is lost among other logs and not really visible in case of failure.

File: src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java (hunk 3 above)
Comment (AnatolyPopov) [reply src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java]: Fixed

File: src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java [hunk 5]
```diff
@@ -297,17 +276,11 @@ void testFailingEvery3rdRequest() throws ExecutionException, InterruptedExceptio
...
-        assertIterableEquals(expectedBodies, bodyRecorderHandler.recorderBodies());
+        await("All expected requests received by HTTP server")
+                .atMost(Duration.ofSeconds(15)).pollInterval(Duration.ofMillis(100))
+                .until(() -> bodyRecorderHandler.recorderBodies().size() >= expectedBodies.size());
 
-        log.info("{} HTTP requests were expected, {} were successfully delivered",
-            expectedBodies.size(),
-            bodyRecorderHandler.recorderBodies().size());
+        assertThat(bodyRecorderHandler.recorderBodies()).containsExactlyInAnyOrderElementsOf(expectedBodies);
```
Comment (hackaugusto) [inline src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java:283]: FYI, I stopped after the second assert, should probably check all the asserts

Comment (AnatolyPopov) [reply src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java]: Fixed

**Review by ivanyu — APPROVED** [review]
